File: golang/core/easy/merge_strings_alternately/merge_strings_alternately.go
package main

import (
    "fmt"
)

func mergeAlternately(word1 string, word2 string) string {
    merged := ""
    i, j := 0, 0

    for i < len(word1) || j < len(word2) {
        if i < len(word1) {
            merged += string(word1[i])
            i++
        }
        if j < len(word2) {
            merged += string(word2[j])
            j++
        }
    }

    return merged
}

func main() {
    fmt.Println(mergeAlternately("ab", "pqrs"))
}

File: typescript/main/merge_strings_alternately.ts
function mergeAlternately(word1: string, word2: string): string {
    let merged = ""
    let i = 0, j = 0

    while (i < word1.length || j < word2.length) {
        if (i < word1.length) {
            merged += word1[i]
            i++
        }
        if (j < word2.length) {
            merged += word2[j]
            j++
        }
    }
    return merged
}

console.log(mergeAlternately("ab", "pqrs"));
